src/backend/data/feed.js

```diff
@@ -37,7 +44,34 @@ class Feed {
    * Returns a Promise.
    */
   save() {
-    addFeed(this);
+    return addFeed(this);
+  }
+
+  /**
+   * Removes current Feed + associated Posts from the databases.
+   * Returns a Promise
+   */
+  async delete() {
+    await removeFeed(this.id);
+    const posts = await getPosts(0, 0);
+    const filteredPosts = posts.filter(async (post) => (await getPost(post).feed) === this.id);
```

Comment:
I think this should be:

```js
const feedId = this.id;

// Get posts ids
let posts = await getPosts(0, 0);
// Get full data for post ids
posts = await Promise.all(posts.map(id => getPost(id)));
// Filter out posts that aren't for this feed
posts = posts.filter(post => post.feed === feedId);
// Take this reduced set of feeds, and remove from redis + elasict
await Promise.all(posts.map(post => Promise.all([removePost(post.id), deletePost(post.id)])));
```

untested :)

Reply:
Refactored even more

Reply:
This is still wrong.  You can't do `posts.filter(async (post) =>...`, filter is synchronous.